### PR TITLE
Migrate to Node 16 LTS

### PR DESCRIPTION
**Pull request by chris-araman**
The GitHub Actions runner recently [gained support](https://github.com/actions/runner/releases/tag/v2.285.0) for Node 16.

However, it looks like that hasn't yet been deployed to all runners, so I'm leaving this in Draft for the time being.
- ❌ `mxcl/xcodebuild` checks [fail](https://github.com/mxcl/xcodebuild/actions/runs/1527413820)
- ✅ `chris-araman/CombineCloudKit` checks [succeed](https://github.com/chris-araman/CombineCloudKit/runs/4386200173)

I suspect it'll be deployed along a similar timeline to their [changing the default Node version](https://github.com/actions/virtual-environments/issues/4446) (over the next few days).

### CODE REVIEW

**Comment by mxcl** [comment]
Oh wonderful, I need this for @git-things-done/template

**Comment by chris-araman** [comment]
Interestingly, the checks seem to pass in the latest manual run. However, I don't know how to verifiably determine which release of [actions/runner](https://github.com/actions/runner) was used in a given job, or even how to determine whether runner [v2.285.0](https://github.com/actions/runner/releases/tag/v2.285.0) is included in a given [actions/virtual-environments](https://github.com/actions/virtual-environments) image. I'm not finding the runner version in the [macos-11](https://github.com/actions/virtual-environments/blob/releases/macOS-11/20211127/images/macos/macos-11-Readme.md) or [macos-10.15](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md) READMEs.

tl;dr: I'm not sure when it's safe to merge this PR. Any ideas?

**Comment by chris-araman** [comment]
Looks like we're good to go:
https://github.blog/changelog/2021-12-10-github-actions-github-hosted-runners-now-run-node-js-16-by-default/

**Review by mxcl — APPROVED** [review]
wonderful.
